apps/emqx/src/emqx_misc.erl

```diff
@@ -371,6 +375,25 @@ explain_posix(estale) -> "Stale remote file handle";
 explain_posix(exdev) -> "Cross-domain link";
 explain_posix(NotPosix) -> NotPosix.
 
+-spec pmap(fun((A) -> B), list(A)) -> list(B | {error, term()}).
+pmap(Fun, List) when is_function(Fun, 1), is_list(List) ->
+    pmap(Fun, List, ?DEFAULT_PMAP_TIMEOUT).
+
+-spec pmap(fun((A) -> B), list(A), timeout()) -> list(B | {error, term()}).
+pmap(Fun, List, Timeout) when
+    is_function(Fun, 1), is_list(List), is_integer(Timeout), Timeout >= 0
+->
+    Self = self(),
+    Pids = lists:map(
+        fun(El) ->
+            spawn_link(
```

Comment:
in case the caller is trapping exit, this link will contaminate caller's mailbox with EXIT (mostly reason=normal) messages.